### DISCUSSION TITLE
ServiceBus trigger retries and defered messages

### DISCUSSION THREAD

**Discussion by bledogit** [Azure/azure-functions-dotnet-worker#1537] [Feedback]
Hi,

I am Looking into the AzF Worker service bus trigger for the ability to send the message back to the queue to retry later as a measure to overcome temporary issues on the processing of the message.

Using lower level APIs, there is the functionality to defer, or at least bypass the autocomplete functionality to be able to bypass the message abandonment in the case of an exception.  The hope was to use the lock expiration as a method to pace the retry of the message to a later time.

Deferring a message would work similarly.  In both cases the MessageReceiver is needed, which is not available on the worker API. The setting however is in the documentation [Azure Service Bus bindings for Azure Functions](https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-service-bus?pivots=programming-language-csharp&tabs=isolated-process%2Cfunctionsv2%2Cextensionv3)

```
{
    "version": "2.0",
    "extensions": {
        "serviceBus": {
            "autoCompleteMessages": true,
        }
    }
}```

Is there a way to accomplish this?

**Comment by jviau** [comment]
@bledogit we will be working on improved service bus bindings as part of our SDK binding efforts. This should give you access to a `MessageReceiver` equivalent type in the new model.

/cc: @JoshLove-msft

**Comment by SeanFeldman** [comment]
[This milestone](https://github.com/Azure/azure-functions-dotnet-worker/milestone/62) will unblock a lot of this.
